### PR TITLE
add analytics tracking for make offer and view offer buttons

### PR DESCRIPTION
**Pull request by xtina-starr**
The type of this PR is: **Chore**

This PR resolves [PURCHASE-2492]

### Description

This adds analytics tracking for the Make Offer and View Offer buttons.

### PR Checklist (tick all before merging)

- [ ] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [ ] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [ ] I have documented any follow-up work that this PR will require, or it does not require any.
- [ ] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [ ] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434
[PURCHASE-2492]: https://artsyproduct.atlassian.net/browse/PURCHASE-2492

### CODE REVIEW

**Comment by xtina-starr** [comment]
Should I be checking in `Podfile.lock`?

**Comment by iskounen** [comment]
> Should I be checking in Podfile.lock?

I haven't been, but I'm curious too. I keep needing to run `pod install --repo-update` for the app to build. @MounirDhahri do you know what the best-practice is for this?

**Comment by MounirDhahri** [comment]
It's fine if you check in `Podfile.lock`.

> Since we are making sure that everyone is using the same version of `cocoapods` using the Gemfile we know for sure that it's going to change only when necessary.

I do however recommend spending some time to just test if whatever got updated did not break something in the app, just like updating any regular npm package.

**Comment by MounirDhahri** [comment]
You could as well create a PR with only the pod changes like I did here https://github.com/artsy/eigen/pull/4547

**Comment by iskounen** [comment]
Oh, I see. Thanks for sharing the best-practice, Mounir 🙌

**Review by lilyfromseattle — APPROVED** [review]
🎉

**Comment by xtina-starr** [comment]
> You could as well create a PR with only the pod changes like I did here #4547

Ahh this is good to know. Thanks all. 

@erikdstock you okay with merging this?

File: src/lib/Scenes/Inbox/Components/Conversations/OpenInquiryModalButton.tsx [hunk 1]
```diff
@@ -15,12 +17,15 @@ const ShadowSeparator = styled(Separator)`
...
 export const OpenInquiryModalButton: React.FC<OpenInquiryModalButtonProps> = ({ artworkID, conversationID }) => {
+  const { trackEvent } = useTracking()
+
   return (
     <>
       <ShadowSeparator />
       <Flex p={1.5}>
         <Button
           onPress={() => {
+            trackEvent(tappedMakeOffer(conversationID as string))
```
Comment (erikdstock) [inline src/lib/Scenes/Inbox/Components/Conversations/OpenInquiryModalButton.tsx:28]: #minor smell to me that `conversationID` isn't guaranteed to be available in this button. I don't think it needs to be a part of this PR but assuming it is nullable i wonder if we couldn't just assert it is present `(!)` higher up the tree. Again doesn't have to be a part of this PR.

**Review by erikdstock — APPROVED** [review]
Seems good here!

**Pull request merged**